File: 백준/Silver/10866. 덱/덱.cc
#include <deque>
#include <iostream>
#include <string>
using namespace std;

int main() {
	int test;
	cin >> test;

	deque<int> dq;

	string str;

	while (test--) {
		cin >> str;

		int input;

		if (str == "push_front") {
			cin >> input;
			dq.push_front(input);
		}
		else if (str == "push_back") {
			cin >> input;
			dq.push_back(input);
		}
		else if (str == "pop_front") {
			if (dq.empty())
				cout << -1<<'\n';
			else {
				cout << dq.front()<<'\n';
				dq.pop_front();
			}
		}
		else if (str == "pop_back") {
			if (dq.empty())
				cout << -1 << '\n';
			else {
				cout << dq.back() << '\n';
				dq.pop_back();
			}
		}
		else if (str == "size") {
			cout << dq.size() << '\n';
		}
		else if (str == "empty") {
			cout<<dq.empty() << '\n';
		}
		else if (str == "front") {
			if (dq.empty())
				cout << -1 << '\n';
			else 
			cout << dq.front() << '\n';
		}
		else if (str == "back") {
			if (dq.empty())
				cout << -1 << '\n';
			else
			cout<<dq.back() << '\n';
		}
	}
}

File: 백준/Silver/10866. 덱/덱.py
import sys

# map(int,sys.stdin.readline().split())
# data = list(map(int,sys.stdin.readline().split()))

n = int(sys.stdin.readline())

dequeue = []

for i in range(n):
    com = list(sys.stdin.readline().split())

    if com[0]=='push_front':
        dequeue.insert(0,com[1])
    elif com[0]=='push_back':
        dequeue.append(com[1])
    elif com[0]=='pop_front':
        if len(dequeue) == 0:
            print(-1)
        else:
            print(dequeue[0])
            del dequeue[0]
    elif com[0]=='pop_back':
        if len(dequeue) == 0:
            print(-1)
        else:
            print(dequeue[-1])
            del dequeue[-1]
    elif com[0]=='size':
        print(len(dequeue))
    elif com[0]=='empty':
        if len(dequeue) == 0:
            print(1)
        else:
            print(0)
    elif com[0]=='front':
        if len(dequeue) == 0:
            print(-1)
        else:
            print(dequeue[0])
    elif com[0]=='back':
        if len(dequeue) == 0:
            print(-1)
        else:
            print(dequeue[-1])
